### PR TITLE
Move boolean parsing to a separate module

### PR DESCRIPTION
**Pull request by MrCreosote**
# Description of PR purpose/changes

Move the boolean parsing function into a separate module. Ran into import cycles when making other changes. Probably makes more sense to have it in a separate module anyway.

# Jira Ticket / Github Issue #
- [n/a] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
- [X] Tests pass in travis and locally 
- [X] Changes available by spinning up a local test suite

# Dev Checklist:

- [X] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [?] My changes generate no new warnings - too many warnings to realistically check
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [n/a] Any dependent changes have been merged and published in downstream modules
- [X] I have run Black and Flake8 on changed Python Code manually or with git precommit (and the travis build passes)

# Updating Version and Release Notes (if applicable)

- [n/a] [Version has been bumped](https://semver.org/) for each release
- [n/a] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

**Comment by MrCreosote** [comment]
The codacy and codeclimate complaints IMO are not useful or are incorrect

**Review by bio-boris — APPROVED** [review]
lgtm

File: lib/execution_engine2/utils/arg_processing.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    if putative_bool is None:
+        return False
+
+    if isinstance(putative_bool, bool):
+        return putative_bool
+
+    if isinstance(putative_bool, int):
+        return bool(putative_bool)
+
+    if isinstance(putative_bool, str):
```
Comment (bio-boris) [inline lib/execution_engine2/utils/arg_processing.py:25]: Do we care about cases when it's a string not equal to "true" or "false" ? like "tru"?

**Comment by bio-boris** [comment]
LGTM, but we might want to avoid our own function and use json.loads instead for t/f values?

File: lib/execution_engine2/utils/arg_processing.py (hunk 1 above)
Comment (bio-boris) [reply lib/execution_engine2/utils/arg_processing.py:25]: Ah never mind it throws the exception.. What about "1.0", "1", which is likely to come from the json rpc endpoints?

**Review by bio-boris — APPROVED** [review]

File: lib/execution_engine2/sdk/EE2StatusRange.py [hunk 2]
```diff
@@ -187,7 +188,7 @@ def get_sort_order(self, ascending):
         if ascending is None:
             return "+"
         else:
-            if self.sdkmr.parse_bool_from_string(ascending):
+            if parse_bool(ascending):
```
Comment (bio-boris) [inline lib/execution_engine2/sdk/EE2StatusRange.py:191]: we should see what the job browser is sending and see if that still works here

File: lib/execution_engine2/utils/arg_processing.py (hunk 1 above)
Comment (MrCreosote) [reply lib/execution_engine2/utils/arg_processing.py]: Added handling for stringified ints and floats

File: lib/execution_engine2/sdk/EE2StatusRange.py (hunk 2 above)
Comment (MrCreosote) [reply lib/execution_engine2/sdk/EE2StatusRange.py:191]: I can do this if you want, but it seems like the updated function should be able to handle absolutely anything thrown at it at this point. I'm assuming for now it's not necessary, but let me know if you'd still like me to check.

**Comment by MrCreosote** [comment]
> we might want to avoid our own function and use json.loads/json.dumps instead for t/f values?

this is the json processing part of the function:

```
        if isinstance(json.loads(str_or_bool.lower()), bool):
            return json.loads(str_or_bool.lower())
```

For `isinstance(...)` to be true, the value of `str_or_bool.lower()` must be `"true" or "false"`. That means, AFAICT, this statement is equivalent:

```
if str_or_bool.lower() == "true"
    return True
if str_or_bool.lower() == "false"
    return False
```

and avoids loading a relatively heavyweight library, initializing any json processing engine state, etc.

We could eliminate a couple lines of code by making a static map `{"true": True, "false": False}` and checking and returning the map value *shrug*

**Pull request merged**